lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -2377,42 +2383,29 @@ static CanSILFunctionType getNativeSILFunctionType(
     switch (constant ? constant->kind : SILDeclRef::Kind::Func) {
     case SILDeclRef::Kind::Initializer:
     case SILDeclRef::Kind::EnumElement:
-      return getSILFunctionType(TC, context, origType, substInterfaceType,
-                                extInfo, DefaultInitializerConventions(),
-                                ForeignInfo(), origConstant, constant, reqtSubs,
-                                witnessMethodConformance);
+      return getSILFunctionTypeForConventions(DefaultInitializerConventions());
```

Comment:
Another approach I use in similar situations is to factor out the switch into a new function that in this case would return the conventions, and then call getSILFunctionType() with the result of this new function.

Reply:
I tried doing that here, but `Conventions` is an abstract class, so I cannot easily return it, I'd have to do some manual boxing+unboxing.

Reply:
You could return a unique_ptr<Conventions>, but yeah.